src/kubelet/metric/metric.go

```diff
@@ -192,7 +191,11 @@ func fetchVolumeStats(v v1.VolumeStats) (definition.RawMetrics, error) {
 }
 
 // GroupStatsSummary groups specific data for pods, containers and node
-func GroupStatsSummary(statsSummary v1.Summary) (definition.RawGroups, []error) {
+func GroupStatsSummary(statsSummary *v1.Summary) (definition.RawGroups, []error) {
```

Comment:
this makes me think that statsSummary is going to be modified inside the function.

Reply:
@ Rob Pike pls add `const` to go asap thanks

Reply:
>@ Rob Pike pls add const to go asap thanks

I subscribed to https://github.com/golang/go/issues/22876 just yesterday :smile: I think Rust does a good job with mutable ref here.

>this makes me think that statsSummary is going to be modified inside the function.


@gsanchezgavier I understand your concern. However, as far as I know, most of the time you should expect that function will not modify it's parameters, but rather return a mutated copy of the data. Modifying in place is a performance optimization IMO.

Given that this function returns a different format, I think it should be obvious to assume the summary won't be modified.